pkg/autoscaler/metrics/collector.go

```diff
@@ -245,6 +253,30 @@ func (c *MetricCollector) StableAndPanicRPS(key types.NamespacedName, now time.T
 	return s, p, nil
 }
 
+// CandidatesForRemoval returns candidates for removal based on stats collected on the
+// number of requests these candidate pods are processing
+func (c *MetricCollector) CandidatesForRemoval(key types.NamespacedName, readyCount, desiredScale int) ([]string, error) {
+	if readyCount <= desiredScale {
+		return nil, nil
+	}
+	c.collectionsMutex.RLock()
+	defer c.collectionsMutex.RUnlock()
+
+	collection, exists := c.collections[key]
+	if !exists {
+		return nil, ErrNotScraping
+	}
+
+	collection.podTrafficMutex.RLock()
+	defer collection.podTrafficMutex.RUnlock()
+	if len(collection.podTraffic) < readyCount - desiredScale {
+		fmt.Printf("\n\n\n\n TARALOG candidatesforremoval returning %#v under len(collection.podTraffic) < readyCount - desiredScale\n\n\n\n", collection.podTraffic)
+		return collection.podTraffic, nil
+	}
+	fmt.Printf("\n\n\n\n TARALOG candidatesforremoval returning %#v\n\n\n\n", collection.podTraffic[:readyCount-desiredScale])
```

Comment:
should either lose this line (and a few others) or convert it to a proper logger log